### PR TITLE
fix: make code work again

### PR DESCRIPTION
**Pull request by MarcoASilva**
### Description

1. fix complete example code
1. add some styles 
    1. show buttons as a list
    1. turn button color green when pressed

### Motivation

Tried code from MDN and it didn't work

### CODE REVIEW

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md [hunk 1]
```diff
@@ -266,7 +266,8 @@ function updateStatus() {
     scangamepads();
   }
 
-  controllers.forEach((controller, i) => {
+  for (const i in controllers) {
+    const controller = controllers[i];
```
Comment (teoli2003) [inline files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:270]: Never use `for...in` with arrays and objects (as extra methods or properties can be added). While `forEach` doesn't work with objects, it is better to do this.
```suggestion
  controllers.entries().forEach((controller, i) => {
```

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md [hunk 2]
```diff
@@ -282,24 +283,31 @@ function updateStatus() {
 
       const pct = `${Math.round(val * 100)}%`;
       b.style.backgroundSize = `${pct} ${pct}`;
+      b.textContent = pressed ? `Button ${i} [PRESSED]` : `Button ${i}`;
+      b.style.color = pressed ? "#42f593" : "#2e2d33";
```
Comment (teoli2003) [inline files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:287]: As we set a class name on the next line, this should be in the CSS.

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md [hunk 3]
```diff
@@ -282,24 +283,31 @@ function updateStatus() {
...
 function scangamepads() {
   const gamepads = navigator.getGamepads();
+  document.querySelector("#noDevices").style.display = gamepads.filter(Boolean)
+    .length
+    ? "none"
+    : "block";
   for (const gamepad of gamepads) {
-    if (gamepad) { // Can be null if disconnected during the session
+    if (gamepad) {
+      // Can be null if disconnected during the session
```
Comment (teoli2003) [inline files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:310]:
```suggestion
    // gamepad is null if disconnected during the session
    if (gamepad) {
```

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md [hunk 4]
```diff
@@ -282,24 +283,31 @@ function updateStatus() {
...
 
   requestAnimationFrame(updateStatus);
 }
 
 function scangamepads() {
   const gamepads = navigator.getGamepads();
+  document.querySelector("#noDevices").style.display = gamepads.filter(Boolean)
+    .length
+    ? "none"
+    : "block";
```
Comment (teoli2003) [inline files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:307]: What is this for?

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md [hunk 5]
```diff
@@ -313,7 +321,8 @@ window.addEventListener("gamepadconnected", connecthandler);
 window.addEventListener("gamepaddisconnected", disconnecthandler);
 
 if (!haveEvents) {
- setInterval(scangamepads, 500);
+  document.querySelector("#noDevices").style.display = "block";
```
Comment (teoli2003) [inline files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:324]: What is this for?

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Yes, the `forEach` doesn't work on an object. I have a few questions on some other changes.

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md (hunk 2 above)
Comment (MarcoASilva) [reply files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:287]: I didn't find a css code snippet on this mdn page, where css class defs should be placed?

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md (hunk 4 above)
Comment (MarcoASilva) [reply files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:307]: Displaying a text to the user informing no devices were found

Comment (MarcoASilva) [reply files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:307]: Up to now nothing is showed (empty/blank page) if no controllers are connected or no buttons were pressed

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md (hunk 5 above)
Comment (MarcoASilva) [reply files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md]: Just removed, as L304 is showing a text to the user when no gamepads were found

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md (hunk 1 above)
Comment (MarcoASilva) [reply files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md]: As controllers is an object (map of gamepads by id) I used `Object.entries`

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md (hunk 2 above)
Comment (teoli2003) [reply files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:303]: Yes, as the CSS is not here, let's keep this that way.

File: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md (hunk 4 above)
Comment (teoli2003) [reply files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md:323]: 👍

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**